src/main/java/com/google/sps/servlets/MealServlet.java

```diff
@@ -84,13 +77,30 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         response.sendError(HttpServletResponse.SC_NOT_FOUND);
         return;
     }
+    /**
+     * create entity for Datastore with properties of class Meal
+     * @param meal
+     * @return
+     */
+    private Entity createMealEntity(Meal meal) {
+        Entity mealEntity = new Entity("Meal");
+        mealEntity.setProperty("id", meal.getId());
+        mealEntity.setProperty("title", meal.getTitle());
+        mealEntity.setProperty("description", meal.getDescription());
+        mealEntity.setProperty("ingredients", meal.getIngredients());
+        mealEntity.setProperty("type", meal.getType());
+ 
+        return mealEntity;
+    }
 
-    private void searchMeal(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        // Fuction returns hardcoded result
-        // TODO(sandatsian): implement search algorithm here for MVP
-        String gson = new Gson().toJson(dishes);
-        response.setContentType("application/json");
-        response.getWriter().println(gson);
+    private void getMealList(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        Query query = new Query("Meal").addSort("id", SortDirection.DESCENDING);
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = datastore.prepare(query);
+        HashMap<Long, Meal> meals = getDataFromDatastore(results);
+        Gson gson = new Gson();
+        response.setContentType("application/json;");
+        response.getWriter().println(gson.toJson(meals));
```

Comment:
You can use `print` instead of `println` if additional end of line in the content is a problem.